src/apps/Altinn.Authorization.DeployApi/src/Altinn.Authorization.DeployApi/BootstrapDatabase/BootstrapDatabasePipeline.cs

```diff
@@ -0,0 +1,138 @@
+using Azure.ResourceManager;
+using Azure.ResourceManager.KeyVault;
+using Azure.ResourceManager.PostgreSql.FlexibleServers;
+using Azure.Security.KeyVault.Secrets;
+using Npgsql;
+using System.Text.Json.Serialization;
+
+namespace Altinn.Authorization.DeployApi.BootstrapDatabase;
+
+internal sealed class BootstrapDatabasePipeline
+    : Pipeline
+{
+    [JsonPropertyName("resources")]
+    public required ResourcesConfig Resources { get; init; }
+
+    [JsonPropertyName("databaseName")]
+    public required string DatabaseName { get; init; }
+
+    [JsonPropertyName("schemas")]
+    public required IReadOnlyDictionary<string, SchemaBootstrapModel> Schemas { get; init; }
+
+    protected internal override async Task ExecuteAsync(PipelineContext context, CancellationToken cancellationToken)
+    {
+        var cred = context.GetRequiredService<TokenCredential>();
+        var client = new ArmClient(cred, defaultSubscriptionId: Resources.SubscriptionId);
+
+        var subscription = await context.RunTask(
+            "Get subscription info",
+            (_, ct) => client.GetDefaultSubscriptionAsync(ct),
+            cancellationToken);
+
+        var resourceGroup = await context.RunTask(
+            "Get resource group info",
+            (_, ct) => subscription.GetResourceGroupAsync(Resources.ResourceGroupName, ct),
+            cancellationToken);
+
+        var keyVault = await context.RunTask(
+            "Get key vault info",
+            (_, ct) => resourceGroup.Value.GetKeyVaultAsync(Resources.KeyVaultName, ct),
+            cancellationToken);
+
+        var server = await context.RunTask(
+            "Get server info",
+            (_, ct) => resourceGroup.Value.GetPostgreSqlFlexibleServerAsync(Resources.ServerName, ct),
+            cancellationToken);
+
+        var token = await context.RunTask(
+            "Get db auth token",
+            (_, ct) => cred.GetTokenAsync(new(["https://ossrdbms-aad.database.windows.net/.default"]), ct).AsTask(),
+            cancellationToken);
+
+        var secretClient = new SecretClient(keyVault.Value.Data.Properties.VaultUri, cred);
+
+        var serverUrl = server.Value.Data.FullyQualifiedDomainName;
+        var connStringBuilder = new NpgsqlConnectionStringBuilder()
+        {
+            Host = serverUrl,
+            Database = "postgres",
+            Username = Resources.User,
+            Password = token.Token,
+            Port = 5432,
+            SslMode = SslMode.Require,
+            Pooling = false,
+        };
+
+        var serverConnString = connStringBuilder.ToString();
+        await using var serverConn = new NpgsqlConnection(serverConnString);
+        await context.RunTask(
+            "Connecting to database server",
+            (_, ct) => serverConn.OpenAsync(ct),
+            cancellationToken);
+
+        var migratorUser = await context.RunTask(new CreateDatabaseRoleTask(secretClient, serverConn, $"{DatabaseName}_migrator", Resources.User), cancellationToken);
```

Comment:
There have been talks if an .NET app should not have their own DB and that API's should have their own schema and share database - just like the situation is between RR and Access mgmt. However, I do prefer the 1 one to 1 mapping as you try to generalize here.

Reply:
I'm not enforcing that at all - this allows databases to be shared, but also to be re-used if that's what you want.